config/tf_modules/aws-s3/files.tf

```diff
@@ -0,0 +1,8 @@
+resource "aws_s3_bucket_object" "user_files" {
+  for_each     = local.files_to_upload
+  bucket       = aws_s3_bucket.bucket.id
+  key          = each.key
+  source       = "${var.files}/${each.key}"
+  content_type = lookup(local.mime_types, regex("\\.[^.]+$", each.key), "binary/octet-stream")
```

Comment:
ooh very cool. This would avoid so many problems for the users!
I've had to debug these mime problems in s3 countless times.